webapp/src/components/contentBlock.tsx

```diff
@@ -168,24 +90,64 @@ class ContentBlock extends React.PureComponent<Props, State> {
                         </MenuWrapper>
                     }
                 </div>
-                {block.type === 'text' &&
-                    <MarkdownEditor
-                        text={block.title}
-                        placeholderText={intl.formatMessage({id: 'ContentBlock.editText', defaultMessage: 'Edit text...'})}
-                        onBlur={(text) => {
-                            mutator.changeTitle(block, text, intl.formatMessage({id: 'ContentBlock.editCardText', defaultMessage: 'edit card text'}))
-                        }}
-                        readonly={this.props.readonly}
-                    />}
-                {block.type === 'divider' && <div className='divider'/>}
-                {block.type === 'image' && this.state.imageDataUrl &&
-                    <img
-                        src={this.state.imageDataUrl}
-                        alt={block.title}
-                    />}
+                <ContentElement
+                    block={block}
+                    readonly={readonly}
+                />
             </div>
         )
     }
+
+    private addContentMenu(type: BlockTypes): JSX.Element {
+        const {intl, card, contents, block} = this.props
+        const index = contents.indexOf(block)
+
+        switch (type) {
+        case 'image': return (
```

Comment:
Looks like every new content that needs a bit of special treatment end ups needed an entry here. I would expect the `onClick` to be handled somehow as part of the content definition.

Reply:
Same for the icon.

Reply:
I think you are oversimplifing it saying that a new content type is just a Component. That is only the rendering part, I would love to have the whole thing self-contained. That would help to allow even add plugins completely client side.

Reply:
Agree that's the direction. Roughly encapsulating that in "RegistryEntry" right now. In your mind:
* Would the plugins generate React components for display? What's the alternative?
* Would they handle the onClicks, or provide an interface of actions (create, update, etc.)?

Have to think more about what's needed for fully dynamic extensions via plugins. E.g. namespacing. In theory, a plugin would implement (a future iteration of) RegistryEntry.

Reply:
Refactored this based on your comments.